src/compiler/crystal/syntax/parser.cr

```diff
@@ -103,7 +103,7 @@ module Crystal
         break if end_token?
       end
 
-      Expressions.from(exps)
+      Expressions.from(exps).at(exps.first)
```

Comment:
The specs seem to get fixed with this patch:

```diff
diff --git a/src/compiler/crystal/syntax/ast.cr b/src/compiler/crystal/syntax/ast.cr
index 687d6df..1cd9538 100644
--- a/src/compiler/crystal/syntax/ast.cr
+++ b/src/compiler/crystal/syntax/ast.cr
@@ -138,6 +138,10 @@ module Crystal
       @expressions.last
     end
 
+    def location
+      @location || @expressions.first?.try &.location
+    end
+
     def end_location
       @end_location || @expressions.last?.try &.end_location
     end
diff --git a/src/compiler/crystal/syntax/parser.cr b/src/compiler/crystal/syntax/parser.cr
index 127ab72..1d3883e 100644
--- a/src/compiler/crystal/syntax/parser.cr
+++ b/src/compiler/crystal/syntax/parser.cr
@@ -103,7 +103,7 @@ module Crystal
         break if end_token?
       end
 
-      Expressions.from(exps).at(exps.first)
+      Expressions.from(exps)
     end
 
     def parse_multi_assign
```

Expressions already had an overload for `end_location` that returned that of the last expression, but we were missing one for `location` as well.

I didn't check if it still fixed the missing `!dbg` issue, but I guess it does.